### PR TITLE
added qr scanner code

### PR DESCRIPTION
**Pull request by YuYing-Liang**
## Overview

- adds qr code scanning to makeuoft.ca
- scanner only appears when admins login

## Unit Tests Created

- none for now

## Steps to QA

- run migrations, collectstatic and compile scss
- as a participant, take a photo of ur qrcode when you login
- login as an admin, go to qr scanner page and scan ur participant code with camera
- might need to change times in settings/__init__.py
- check for errors:
- throws an error for duplicate sign-ins
- participant who hasn't RSVP'd and isn't accepted or a user that doesn't exist
- sign in data should appear on django admin site (model called UserActivity)
- test for general user experience and fluidity

### CODE REVIEW

File: hackathon_site/event/static/event/styles/scss/_variables.scss [hunk 1]
```diff
@@ -11,6 +11,16 @@ $colors: (
     secondaryOnHover: #d2a300,
     errorOnHover: #870000,
     successOnHover: #1d7521,
+
+    backgroundColor: #ddeeff,
+    error: #b00020,
```
Comment (Mustaballer) [inline hackathon_site/event/static/event/styles/scss/_variables.scss]: Remove duplicate keys for error and success. Plus run prettier to pass template checks

File: hackathon_site/hackathon_site/settings/__init__.py [hunk 2]
```diff
@@ -352,3 +354,33 @@
...
 # HSS Testing
 TEST_USER_GROUP = "HSS Test Users"
+
+# sign in times must be between EVENT_START_DATE and EVENT_END_DATE and in chronological order
+# the number of sign in times MUST MATCH the number of columns in UserActivityTable
+SIGN_IN_TIMES = [
+    {
+        "name": "sign_in",
+        "description": "Hackathon Sign In",
+        "time": datetime(2023, 1, 13, 17, 0, 0, tzinfo=TZ_INFO),  # Oct 10th @ 11am
```
Comment (Mustaballer) [inline hackathon_site/hackathon_site/settings/__init__.py]: We should adjust the times of the events to match the participant package

**Review by Mustaballer — CHANGES_REQUESTED** [review]
One suggestion to change 'User activitys' to 'User activities' in Django Admin Site.

When I sign in multiple times(works first time), I get the following error. Is that what we expect?
image

When I try to sign in an email that doesn't exist it lets me know. When I input an email that does exist, and didn't apply, it lets me know with correct message.  If I don't rsvp, I don't get access to the QR code as a participant. ✔

**Review by Mustaballer — COMMENTED** [review]
It works wonderfully! I tested signing in multiple times, and it was able to track that and gave me the correct error. When changing events, there were no issues with signing in again. Before your merge the PR, you should run prettier to pass the template checks, and on a final note to update the last two event times to match the participant package :)

File: hackathon_site/event/jinja2/event/admin_qr_scanner.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+{% endblock %}
+
+{% block scripts %}
+    <script src="{{ static('event/js/qr-scanner.umd.min.js') }}"></script>
+    <script type="text/javascript">
+        const videoElem = document.getElementById("scanner");
+        let oldData = '';
+
+        $("#signInButton").click(() => {
+            console.log(oldData.split(';'))
```
Comment (Mustaballer) [inline hackathon_site/event/jinja2/event/admin_qr_scanner.html]: Remove log statement

**Review by Mustaballer — APPROVED** [review]
Great job!

**Review by Leo6Leo — CHANGES_REQUESTED** [review]
Temporary Merging Blocking

**Pull request merged**